python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -193,7 +62,11 @@ public Call(SubscriptionContext ctx, CallExpression callExpression) {
     }
 
     boolean hasArgument(String name, Predicate<Expression> predicate) {
-      return getArgument(ctx, callExpression, name).filter(flow -> flow.hasExpression(predicate)).isPresent();
+      return hasArgument(name, -1, predicate);
```

Comment:
why `-1`?

Reply:
I use -1 as an invalid/impossible position, in case the concerned CallExpression is about a python method and an argument which can only be passed by name. In that case there is no valid position, so it should only retrieve the argument if it can find a valid name.

With a bit of insight now it does not look great :D Maybe should I replace `int` with `Integer`, so I can pass `null` value instead, but I would also need to modify `getArgument()` to support the `null` value and not call `TreeUtils.nthArgumentOrKeyword()` in that case, as it is expecting a `int` as `argPos`. Or I can also adapt `TreeUtils.nthArgumentOrKeyword()` to use `Integer` and support null value.

WDYT ?

Reply:
I see, thanks for clarification! I would be a bit more explicit and have a dedicated `hasKeywordArgument` and probably use `TreeUtils.argumentByKeyword` utility